framework/classes/orm/behaviour/virtualname.php

```diff
@@ -105,6 +105,18 @@ public function virtual_name(\Nos\Orm\Model $item, $virtual_name = null)
         return $item->{$this->_properties['virtual_name_property']};
     }
 
+    public function virtual_name_populate_value(\Nos\Orm\Model $item)
+    {
+        if (!empty($this->_properties['populate_property'])) {
+            if (isset($item->{$this->_properties['populate_property']})) {
+                return $item->{$this->_properties['populate_property']};
+            } elseif (method_exists($item, $this->_properties['populate_property'])) {
```

Comment:
Would have done the other way around : give priority to method and then to property.